content/Bias_variance - Mikaela pisani.md

```diff
@@ -0,0 +1,112 @@
+
+## How to diagnose bias/variance error by looking at Learning curves? 
+
+
+### Plot error against the amount of training examples 
+You would need to run the algorithm several times, changing the training set size. As the training set size increases, the error should decrease.
+
+<img src="images/plot_trainingsize_error.png"
+     alt="Plot error vs training size"
+     width="500" height="250"
+     align="middle"/>
+     
+You can figure out how much closer you can get to the desired performance “optimal error rate” by adding more data. There is some point when adding more data does not improve the performance. Graphically you can notice this point when the error curve is plateaued, it means that the error states the same.
+
+The training error can also help you to understand the point when adding more data is useless. 
+Your validation error should decrease as the training increases. But your training set error usually ​increases​ as the training set size grows. But, as we can see in the graph, It also grows up to a point. There is a point where the validation and training error do not change when adding more data. It means that from that point, it does not matter how much data we add, we cannot improve the error performance of the algorithm. 
+
+<img src="images/plot_bias_variance_trainingsize.png"
+     alt="Plot error vs training size - bias_variance"
+     width="500" height="250"
+     align="middle"/>
+
+### Plot error against the number of iterations 
+The learning curve can have in the x-axis either the amount of elements in the training set or the amount of iterations. In the examples above the x-axis was the training set size. In the following curve we can see that there is also a point when the error does not improve even if we increase the number of iterations. 
+
+<img src="images/plot_iterations_error.png"
+     alt="Plot error vs number of iterations"
+     width="500" height="250"
+     align="middle"/>
+     
+ Plotting this curve we can also detect overfitting. As we can see in the following graph, it happens when increasing the number of iterations the training error goes down but the validation error starts to increase. I recommend you to read [this](https://machinelearningmastery.com/learning-curves-for-diagnosing-machine-learning-model-performance/) blog for further information about the detection of overfitting from this kind of plot.
+
+<img src="images/plot_iterations_overfitting.png"
+     alt="Plot error vs number of iterations - overfitting"
+     width="500" height="250"
+     align="middle"/>
+
+### How can we reduce bias? How can we reduce variance?  
+
+**Increase the complexity of the model:** For example, in a neural network you can change the number of nodes or layers. By increasing the complexity of your model you can reduce the bias. On the other hand, this might present computational problems and also can lead to an increase of variance, and as a consequence possibility of overfitting. 
```

Comment:
```suggestion
**Increase the complexity of the model:** For example, in a neural network, you can change the number of nodes or layers. By increasing the complexity of your model you can reduce the bias. On the other hand, this might present computational problems and also can lead to an increase of variance, and as a consequence possibility of overfitting. 
```